waku/v2/node/wakunode2.nim

```diff
@@ -130,35 +130,46 @@ template tcpEndPoint(address, port): auto =
 template addWsFlag() =
   MultiAddress.init("/ws").tryGet()
 
+template addWssFlag() =
+  MultiAddress.init("/wss").tryGet()
+
 
 proc new*(T: type WakuNode, nodeKey: crypto.PrivateKey,
     bindIp: ValidIpAddress, bindPort: Port,
     extIp = none[ValidIpAddress](), extPort = none[Port](),
     peerStorage: PeerStorage = nil,
     maxConnections = builders.MaxConnections,
     wsBindPort: Port = (Port)8000,
-    wsEnabled: bool = false): T 
-    {.raises: [Defect, LPError].} =
+    wsEnabled: bool = false,
+    wssEnabled: bool = false,
+    secureKey: string = "",
+    secureCert: string = ""): T 
+    {.raises: [Defect, LPError, IOError,TLSStreamProtocolError].} =
   ## Creates a Waku Node.
   ##
   ## Status: Implemented.
   ##
   let
     rng = crypto.newRng()
     hostAddress = tcpEndPoint(bindIp, bindPort)
-    wsHostAddress = tcpEndPoint(bindIp, wsbindPort) & addWsFlag
+    wsHostAddress = if wssEnabled: tcpEndPoint(bindIp, wsbindPort) & addWssFlag
+                    else: tcpEndPoint(bindIp, wsbindPort) & addWsFlag
     announcedAddresses = if extIp.isNone() or extPort.isNone(): @[]
-                        elif wsEnabled == false: @[tcpEndPoint(extIp.get(), extPort.get())]
+                        elif wsEnabled == false and wssEnabled == false: 
+                          @[tcpEndPoint(extIp.get(), extPort.get())]
+                        elif wssEnabled:
+                          @[tcpEndPoint(extIp.get(), extPort.get()),
+                          tcpEndPoint(extIp.get(), wsBindPort) & addWssFlag]
                         else : @[tcpEndPoint(extIp.get(), extPort.get()),
-                        tcpEndPoint(extIp.get(), wsBindPort) & addWsFlag]
+                          tcpEndPoint(extIp.get(), wsBindPort) & addWsFlag]
     peerInfo = PeerInfo.new(nodekey)
     enrIp = if extIp.isSome(): extIp
             else: some(bindIp)
     enrTcpPort = if extPort.isSome(): extPort
                  else: some(bindPort)
     enr = createEnr(nodeKey, enrIp, enrTcpPort, none(Port))
   
-  if wsEnabled == true:
+  if wsEnabled == true or wssEnabled == true:
```

Comment:
Can there be a case that `wsEnabled` is false and `wssEnabled` true?

Reply:
If the user choses to do so , we will raise a warning however the logic for wakunode won't change as we will take wss as its the secure version